packages/cln_plugin/lib/src/cln_plugin_base.dart

```diff
@@ -86,8 +89,12 @@ class Plugin implements CLNPlugin {
   }
 
   @override
-  void registerHook({required String name}) {
+  void registerHook(
+      {required String name,
+      required Future<Map<String, Object>> Function(Plugin, Map<String, Object>)
+          callback}) {
     hooks.add(name);
+    hookCallback[name] = RPCHook(name: name, callback: callback);
```

Comment:
Was facing issues with referring to `RPCHook` object, so had to use an extra data structure. Any reviews or a better approach is welcome. :)

Reply:
This is the correct implementation imho, I did not try this in rust but I will do in a while :)

But for now looks so good!